client/theia-glsp-server/src/node/glsp-server-launcher.ts

```diff
@@ -0,0 +1,101 @@
+/********************************************************************************
+ * Copyright (c) 2019 EclipseSource and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+import { ILogger } from '@theia/core';
+import { BackendApplicationContribution } from '@theia/core/lib/node';
+import { ProcessErrorEvent } from '@theia/process/lib/node/process';
+import { ProcessManager } from '@theia/process/lib/node/process-manager';
+import { RawProcess, RawProcessFactory } from '@theia/process/lib/node/raw-process';
+import * as cp from 'child_process';
+import { inject, injectable, optional } from 'inversify';
+
+export const GLSPLaunchOptions = Symbol.for('LaunchOptions');
+export interface GLSPLaunchOptions {
+    isRunning: boolean
+    serverPort: number
+    hostname: string
+    jarPath?: string
+    additionalArgs?: string[];
+}
+
+@injectable()
+export class GLSPServerLauncher implements BackendApplicationContribution {
+    @inject(GLSPLaunchOptions) @optional() protected readonly launchOptions: GLSPLaunchOptions;
+    @inject(RawProcessFactory) protected readonly processFactory: RawProcessFactory;
+    @inject(ProcessManager) protected readonly processManager: ProcessManager;
+    @inject(ILogger) private readonly logger: ILogger;
+
+    initialize() {
+        if (!this.launchOptions.isRunning && !this.start()) {
+            this.logError('Error during model server startup');
+        }
+    }
+
+    start(): boolean {
+        if (!this.launchOptions.isRunning) {
+            return this.startServer();
+        }
+        return true;
+    }
+
+    protected startServer(): boolean {
+        if (this.launchOptions.jarPath) {
+            let args = ['-jar', this.launchOptions.jarPath, '--port', `${this.launchOptions.serverPort}`];
```

Comment:
I'm not sure this `port` passing is working as in https://github.com/eclipsesource/ecore-glsp/blob/8e1b95ee69d3de3ed96c7b2fca32f6ae811827f9/server/ecore-glsp/src/main/java/com/eclipsesource/glsp/ecore/EcoreServerLauncher.java#L29 the port is hardcoded, could you double check?

Reply:
I've adapted the port retrieval. It is now read from commandline arguments. Please have a look :)